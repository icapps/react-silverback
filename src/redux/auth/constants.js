export default {
  LOGIN_USER: 'LOGIN_USER',
  LOGIN_USER_PENDING: 'LOGIN_USER_PENDING',
  LOGIN_USER_REJECTED: 'LOGIN_USER_REJECTED',
  LOGIN_USER_FULFILLED: 'LOGIN_USER_FULFILLED',
  LOGOUT_USER: 'LOGOUT_USER',
};
