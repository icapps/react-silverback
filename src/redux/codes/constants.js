export default {
  GET_LANGUAGE_CODES: 'GET_LANGUAGE_CODES',
  GET_LANGUAGE_CODES_PENDING: 'GET_LANGUAGE_CODES_PENDING',
  GET_LANGUAGE_CODES_FULFILLED: 'GET_LANGUAGE_CODES_FULFILLED',
  GET_LANGUAGE_CODES_REJECTED: 'GET_LANGUAGE_CODES_REJECTED',

  GET_LANGUAGE_CODE_BY_ID: 'GET_LANGUAGE_CODE_BY_ID',
  GET_LANGUAGE_CODE_BY_ID_PENDING: 'GET_LANGUAGE_CODE_BY_ID_PENDING',
  GET_LANGUAGE_CODE_BY_ID_FULFILLED: 'GET_LANGUAGE_CODE_BY_ID_FULFILLED',
  GET_LANGUAGE_CODE_BY_ID_REJECTED: 'GET_LANGUAGE_CODE_BY_ID_REJECTED',

  CREATE_LANGUAGE_CODE: 'CREATE_LANGUAGE_CODE',
  CREATE_LANGUAGE_CODE_PENDING: 'CREATE_LANGUAGE_CODE_PENDING',
  CREATE_LANGUAGE_CODE_FULFILLED: 'CREATE_LANGUAGE_CODE_FULFILLED',
  CREATE_LANGUAGE_CODE_REJECTED: 'CREATE_LANGUAGE_CODE_REJECTED',

  DEPRECATE_LANGUAGE_CODE: 'DEPRECATE_LANGUAGE_CODE',
  DEPRECATE_LANGUAGE_CODE_PENDING: 'DEPRECATE_LANGUAGE_CODE_PENDING',
  DEPRECATE_LANGUAGE_CODE_FULFILLED: 'DEPRECATE_LANGUAGE_CODE_FULFILLED',
  DEPRECATE_LANGUAGE_CODE_REJECTED: 'DEPRECATE_LANGUAGE_CODE_REJECTED',

  UNDEPRECATE_LANGUAGE_CODE: 'UNDEPRECATE_LANGUAGE_CODE',
  UNDEPRECATE_LANGUAGE_CODE_PENDING: 'UNDEPRECATE_LANGUAGE_CODE_PENDING',
  UNDEPRECATE_LANGUAGE_CODE_FULFILLED: 'UNDEPRECATE_LANGUAGE_CODE_FULFILLED',
  UNDEPRECATE_LANGUAGE_CODE_REJECTED: 'UNDEPRECATE_LANGUAGE_CODE_REJECTED',

  GET_STATUS_CODES: 'GET_STATUS_CODES',
  GET_STATUS_CODES_PENDING: 'GET_STATUS_CODES_PENDING',
  GET_STATUS_CODES_REJECTED: 'GET_STATUS_CODES_REJECTED',
  GET_STATUS_CODES_FULFILLED: 'GET_STATUS_CODES_FULFILLED',

  SET_CODES_SORT: 'SET_CODES_SORT',
};
