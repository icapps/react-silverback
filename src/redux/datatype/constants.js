export default {
  GET_DATA_TYPE: 'GET_DATA_TYPE',
  GET_DATA_TYPE_PENDING: 'GET_DATA_TYPE_PENDING',
  GET_DATA_TYPE_REJECTED: 'GET_DATA_TYPE_REJECTED',
  GET_DATA_TYPE_FULFILLED: 'GET_DATA_TYPE_FULFILLED',
};
