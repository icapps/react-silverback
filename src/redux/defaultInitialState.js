const defaultInitialState = {
  isPending: false,
  isError: false,
  errorMessage: '',
};

export default defaultInitialState;
