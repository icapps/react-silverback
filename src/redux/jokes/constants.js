export default {
  GET_JOKE           : 'GET_JOKE',
  GET_JOKE_REQUEST   : 'GET_JOKE_REQUEST',
  GET_JOKE_FULFILLED : 'GET_JOKE_FULFILLED',
  GET_JOKE_REJECTED  : 'GET_JOKE_REJECTED',
};
