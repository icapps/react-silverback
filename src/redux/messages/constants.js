export default {
  SET_MESSAGE: 'SET_MESSAGE',
  
  CLEAR_MESSAGES: 'CLEAR_MESSAGES',
};
