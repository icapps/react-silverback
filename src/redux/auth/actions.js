import constants from './constants';
import { authService } from '../../services';

export const loginUser = (email, password) => {
  return {
    type: constants.LOGIN_USER,
    payload: authService.login(email, password),
  };
};

export const getMe = () => {
  return {
    type: constants.GET_ME,
    payload: authService.getMe(),
  };
};

export const logoutUser = () => {
  return {
    type: constants.LOGOUT_USER,
    payload: authService.logout(),
  };
};

export const forgotPassword = email => {
  return {
    type: constants.FORGOT_PASSWORD,
    payload: authService.forgotPassword(email),
  };
};

export const verifyForgotPassword = token => {
  return {
    type: constants.VERIFY_FORGOT_PASSWORD,
    payload: authService.verifyForgotPassword(token),
  };
};

export const confirmForgotPassword = (token, password) => {
  return {
    type: constants.CONFIRM_FORGOT_PASSWORD,
    payload: authService.confirmForgotPassword(token, password),
  };
};
