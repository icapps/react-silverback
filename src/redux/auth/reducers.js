import constants from './constants';
import defaultInitialState from '../defaultInitialState';
import { strings } from '../../utils';

const initialState = {
  isLoggedIn: false,
  isPasswordVerified: false,
  isPasswordConfirmed: false,
  ...defaultInitialState,
};

const errorMessageHandling = payload => payload.errors[0].title || strings.GENERAL_ERROR;
const auth = (state = initialState, action = {}) => {
  const { payload } = action;

  switch (action.type) {
    case constants.LOGIN_USER_FULFILLED:
      localStorage.setItem('ACCESS_TOKEN', payload.data.accessToken);
      localStorage.setItem('REFRESH_TOKEN', payload.data.refreshToken);
      return {
        ...state,
        isLoggedIn: true,
        isPending: false,
      };
    case constants.LOGIN_USER_PENDING:
      return {
        ...state,
        isLoggedIn: false,
        isPending: true,
        isError: false,
        errorMessage: '',
      };
    case constants.LOGIN_USER_REJECTED:
      return {
        ...state,
        isPending: false,
        isError: true,
        errorMessage: errorMessageHandling(payload),
      };
    // VERIFY_FORGOT_PASSWORD
    case constants.VERIFY_FORGOT_PASSWORD_FULFILLED:
      return {
        ...state,
        isPasswordVerified: true,
        isPending: false,
      };
    case constants.VERIFY_FORGOT_PASSWORD_PENDING:
      return {
        ...state,
        isPasswordVerified: false,
        isPending: true,
        isError: false,
        errorMessage: '',
      };
    case constants.VERIFY_FORGOT_PASSWORD_REJECTED:
      return {
        ...state,
        isPending: false,
        isError: true,
        errorMessage: errorMessageHandling(payload),
      };
    // CONFIRM_FORGOT_PASSWORD
    case constants.CONFIRM_FORGOT_PASSWORD_FULFILLED:
      return {
        ...state,
        isPasswordConfirmed: true,
        isPending: false,
      };
    case constants.CONFIRM_FORGOT_PASSWORD_PENDING:
      return {
        ...state,
        isPasswordConfirmed: false,
        isPending: true,
        isError: false,
        errorMessage: '',
      };
    case constants.CONFIRM_FORGOT_PASSWORD_REJECTED:
      return {
        ...state,
        isPending: false,
        isError: true,
        errorMessage: errorMessageHandling(payload),
      };
    // LOGOUT_USER
    case constants.LOGOUT_USER:
      return initialState;
    default:
      return state;
  }
};

export default auth;
