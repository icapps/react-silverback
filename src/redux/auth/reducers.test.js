import auth from './reducers';
import constants from './constants';
import defaultInitialState from '../defaultInitialState';

const initialState = {
  isLoggedIn: false,
  ...defaultInitialState,
};

describe('auth reducer', () => {
  it('should return the initial state', () => {
    expect(auth(undefined, {})).toMatchSnapshot();
  });

  it('should LOGIN_USER_FULFILLED', () => {
    expect(
      auth(initialState, {
        type: constants.LOGIN_USER_FULFILLED,
        payload: { data: {}, meta: {} },
      }),
    ).toMatchSnapshot();
  });
  it('should LOGIN_USER_PENDING', () => {
    expect(
      auth(initialState, {
        type: constants.LOGIN_USER_PENDING,
      }),
    ).toMatchSnapshot();
  });
  //FORGOT_PASSWORD
  it('should FORGOT_PASSWORD_FULFILLED', () => {
    expect(
      auth(initialState, {
        type: constants.FORGOT_PASSWORD_FULFILLED,
        payload: { data: {}, meta: {} },
      }),
    ).toMatchSnapshot();
  });
  it('should FORGOT_PASSWORD_PENDING', () => {
    expect(
      auth(initialState, {
        type: constants.FORGOT_PASSWORD_PENDING,
      }),
    ).toMatchSnapshot();
  });
  it('should FORGOT_PASSWORD_REJECTED', () => {
    expect(
      auth(initialState, {
        type: constants.FORGOT_PASSWORD_REJECTED,
        payload: { errors: [{ title: 'error' }] },
      }),
    ).toMatchSnapshot();
  });
  //VERIFY_FORGOT_PASSWORD
  it('should VERIFY_FORGOT_PASSWORD_FULFILLED', () => {
    expect(
      auth(initialState, {
        type: constants.VERIFY_FORGOT_PASSWORD_FULFILLED,
        payload: { data: {}, meta: {} },
      }),
    ).toMatchSnapshot();
  });
  it('should VERIFY_FORGOT_PASSWORD_PENDING', () => {
    expect(
      auth(initialState, {
        type: constants.VERIFY_FORGOT_PASSWORD_PENDING,
      }),
    ).toMatchSnapshot();
  });
  it('should VERIFY_FORGOT_PASSWORD_REJECTED', () => {
    expect(
      auth(initialState, {
        type: constants.VERIFY_FORGOT_PASSWORD_REJECTED,
        payload: { errors: [{ title: 'error' }] },
      }),
    ).toMatchSnapshot();
  });
  //CONFIRM_FORGOT_PASSWORD
  it('should CONFIRM_FORGOT_PASSWORD_FULFILLED', () => {
    expect(
      auth(initialState, {
        type: constants.CONFIRM_FORGOT_PASSWORD_FULFILLED,
        payload: { data: {}, meta: {} },
      }),
    ).toMatchSnapshot();
  });
  it('should CONFIRM_FORGOT_PASSWORD_PENDING', () => {
    expect(
      auth(initialState, {
        type: constants.CONFIRM_FORGOT_PASSWORD_PENDING,
      }),
    ).toMatchSnapshot();
  });
  it('should CONFIRM_FORGOT_PASSWORD_REJECTED', () => {
    expect(
      auth(initialState, {
        type: constants.CONFIRM_FORGOT_PASSWORD_REJECTED,
        payload: { errors: [{ title: 'error' }] },
      }),
    ).toMatchSnapshot();
  });
  //LOGOUT_USER
  it('should LOGOUT_USER', () => {
    expect(
      auth(initialState, {
        type: constants.LOGOUT_USER,
      }),
    ).toMatchSnapshot();
  });
});
