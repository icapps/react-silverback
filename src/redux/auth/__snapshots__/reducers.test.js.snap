// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth reducer should CONFIRM_FORGOT_PASSWORD_FULFILLED 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isLoggedIn": false,
  "isPasswordConfirmed": true,
  "isPending": false,
}
`;

exports[`auth reducer should CONFIRM_FORGOT_PASSWORD_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isLoggedIn": false,
  "isPasswordConfirmed": false,
  "isPending": true,
}
`;

exports[`auth reducer should CONFIRM_FORGOT_PASSWORD_REJECTED 1`] = `
Object {
  "errorMessage": "error",
  "isError": true,
  "isLoggedIn": false,
  "isPending": false,
}
`;

exports[`auth reducer should FORGOT_PASSWORD_FULFILLED 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isLoggedIn": false,
  "isPending": false,
}
`;

exports[`auth reducer should FORGOT_PASSWORD_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isLoggedIn": false,
  "isPending": true,
}
`;

exports[`auth reducer should FORGOT_PASSWORD_REJECTED 1`] = `
Object {
  "errorMessage": "error",
  "isError": true,
  "isLoggedIn": false,
  "isPending": false,
}
`;

exports[`auth reducer should LOGIN_USER_FULFILLED 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isLoggedIn": true,
  "isPending": false,
}
`;

exports[`auth reducer should LOGIN_USER_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isLoggedIn": false,
  "isPending": true,
}
`;

exports[`auth reducer should LOGIN_USER_REJECTED 1`] = `
Object {
  "errorMessage": "test",
  "isError": true,
  "isLoggedIn": false,
  "isPending": false,
}
`;

exports[`auth reducer should LOGOUT_USER_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isLoggedIn": false,
  "isPasswordConfirmed": false,
  "isPasswordVerified": false,
  "isPending": false,
}
`;

exports[`auth reducer should VERIFY_FORGOT_PASSWORD_FULFILLED 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isLoggedIn": false,
  "isPasswordVerified": true,
  "isPending": false,
}
`;

exports[`auth reducer should VERIFY_FORGOT_PASSWORD_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isLoggedIn": false,
  "isPasswordVerified": false,
  "isPending": true,
}
`;

exports[`auth reducer should VERIFY_FORGOT_PASSWORD_REJECTED 1`] = `
Object {
  "errorMessage": "error",
  "isError": true,
  "isLoggedIn": false,
  "isPending": false,
}
`;

exports[`auth reducer should return the initial state 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isLoggedIn": false,
  "isPasswordConfirmed": false,
  "isPasswordVerified": false,
  "isPending": false,
}
`;
